tests/testthat/test-plot_with_settings.R

```diff
@@ -58,3 +58,468 @@ testthat::test_that("if teal.plot_dpi is not an integer value then get_plot_dpi
+    ), label = "Unique Subject Identifier"),
+    AGE = structure(c(
+      40L, 24L, 40L, 28L, 37L,
+      40L, 40L, 38L, 34L, 45L, 40L, 24L, 40L, 28L, 37L,
+      40L, 40L, 38L, 34L, 45L
+    ), label = "Age"),
+    BMRKR1 = structure(c(
+      14.424933692778,
+      4.05546277230382, 2.80323956920649, 10.2627340069523, 6.2067627167943,
+      6.9067988141075, 0.463560441314472, 2.85516419937308, 4.99722573047567,
+      4.57499101339464, 11.1444469908374, 2.86312402599659, 7.20634823208459,
+      2.82014082273392, 6.04894627754598, 6.61366727233138, 7.01629609277248,
+      9.00518541690504, 3.68284941334409, 7.09845214575852
+    ), label = "Continuous Level Biomarker 1")
+  ),
+  row.names = c(NA, -20L),
+  label = "Subject Level Analysis Dataset",
+  "`creation date`" = structure(19307, class = "Date"),
+  class = c("tbl_df", "tbl", "data.frame")
+)
+
+
+brush <- list(
+  xmin = 30.461738950382, xmax = 41.763695852997, ymin = 0.73077938561424, ymax = 10.23846022551,
+  coords_css = list(
+    xmin = 511.458312988281,
+    xmax = 1231.45831298828, ymin = 195.642349243164, ymax = 535.642349243164
+  ),
+  coords_img = list(
+    xmin = 460.312444477212, xmax = 1108.31239209208,
+    ymin = 176.078114318848, ymax = 482.078114318848
+  ),
+  img_css_ratio = list(
+    x = 0.899999927242866, y = 0.9
+  ),
+  mapping = list(
+    x = "AGE",
+    y = "BMRKR1"
+  ),
+  domain = list(
+    left = 22.95, right = 46.05,
+    bottom = -0.234508221258705, top = 15.1230023553512
+  ),
+  range = list(
+    left = 29.6253972440434, right = 1354.06849315068,
+    bottom = 513.145417973348, top = 18.8715445043173
+  ),
+  log = list(
+    x = NULL, y = NULL
+  ),
+  direction = "xy",
+  brushId = "teal-main_ui-root-scatterplot_choices-scatter_plot-plot_brush",
+  outputId = "teal-main_ui-root-scatterplot_choices-scatter_plot-plot_main"
+)
+
+testthat::test_that("clean_brushedPoints assert error", {
+  testthat::expect_error(clean_brushedPoints())
+  testthat::expect_error(clean_brushedPoints(2, list()))
+  testthat::expect_error(clean_brushedPoints(data.frame(), list()))
+})
+
+testthat::test_that("clean_brushedPoints minimal correct input", {
+  testthat::expect_error(
+    clean_brushedPoints(
+      data.frame(AGE = 1, BMRKR1 = 4),
+      brush[c("direction", "range", "xmin", "xmax", "ymin", "ymax", "mapping")]
+    ),
+    NA
+  )
+})
+
+testthat::test_that("clean_brushedPoints removal of NA points", {
+  testthat::expect_identical(nrow(data), 20L)
+  testthat::expect_identical(nrow(clean_brushedPoints(data, brush)), 11L)
+  data$AGE[1:10] <- NA
+  testthat::expect_identical(nrow(clean_brushedPoints(data, brush)), 6L)
+})
+
+download_srv_args <- list(
+  id = "STH",
+  plot_reactive = function() plot(1),
+  plot_type = function() "base",
+  plot_w = function() 250,
+  default_w = function() 300,
+  plot_h = function() 200,
+  default_h = function() 300
+)
+
+testthat::test_that("type_download_srv download all types of files with default name", {
```

Comment:
we are not checking for the default name here, we just check the correct type.

Reply:
or actually, shouldnt the default name be the actual time stamp?

Reply:
in testServer inputs are empty, so the name input is empty and the file names is only a file extension suffix.